test/e2e/fabric_contract_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright © 2021 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hyperledger/firefly/pkg/fftypes"
+	gonanoid "github.com/matoous/go-nanoid/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type fabConnectHeaders struct {
+	Type      string `json:"type"`
+	Channel   string `json:"channel"`
+	Signer    string `json:"signer"`
+	Chaincode string `json:"chaincode"`
+}
+
+type createAssetBody struct {
+	Headers fabConnectHeaders `json:"headers"`
+	Func    string            `json:"func"`
+	Args    []string          `json:"args"`
+}
+
+var assetCreatedEvent = &fftypes.FFIEvent{
+	Name: "AssetCreated",
+}
+
+func deployChaincode(t *testing.T, stackName string) string {
+	id, err := gonanoid.New()
+	require.NoError(t, err)
+	chaincodeName := "e2e_" + id
+
+	cmd := exec.Command("bash", "./deploy_chaincode.sh")
+	cmd.Env = append(cmd.Env, "STACK_NAME="+stackName)
+	cmd.Env = append(cmd.Env, "CHAINCODE_NAME="+chaincodeName)
+	cmd.Env = append(cmd.Env, "PATH="+os.Getenv("PATH"))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	require.NoError(t, err)
+
+	return chaincodeName
+}
+
+func invokeFabContract(t *testing.T, client *resty.Client, channel, chaincode, signer, method string, args []string) {
+	path := "/transactions"
+	body := &createAssetBody{
+		Headers: fabConnectHeaders{
+			Type:      "SendTransaction",
+			Channel:   channel,
+			Signer:    signer,
+			Chaincode: chaincode,
+		},
+		Func: method,
+		Args: args,
+	}
+	resp, err := client.R().
+		SetHeader("x-firefly-sync", "true").
+		SetBody(body).
+		Post(path)
+	require.NoError(t, err)
+	require.Equal(t, 200, resp.StatusCode(), "POST %s [%d]: %s", path, resp.StatusCode(), resp.String())
+}
+
+type FabricContractTestSuite struct {
+	suite.Suite
+	testState *testState
+}
+
+func (suite *FabricContractTestSuite) BeforeTest(suiteName, testName string) {
+	suite.testState = beforeE2ETest(suite.T())
+}
+
+func (suite *FabricContractTestSuite) TestE2EContractEvents() {
+	defer suite.testState.done()
+
+	received1, changes1 := wsReader(suite.T(), suite.testState.ws1)
+
+	stack := readStackFile(suite.T())
+	name := deployChaincode(suite.T(), stack.Name)
```

Comment:
I would add a `SetupTestSuite` to the `FabricContractTestSuite` struct and move this there. For IDs or any other things that need to be shared between tests, add them as fields on the `FabricContractTestSuite` struct.